accepted/future-releases/primary-constructors/feature-specification.md

```diff
@@ -702,91 +578,59 @@ constructors as well.
      ('=' <expression>)?;
 ```
 
-A _declaring constructor_ declaration is a declaration that contains a
-`<declaringConstructorSignature>` with a `<declaringParameterList>`, or a
-declaration that contains a `<declaringConstantConstructorSignature>`, or
-it is a `<primaryConstructor>` in the header of a class, enum, or extension
-type declaration, together with a declaration in the body that contains a
-`<declaringConstructorSignature>` *(which does not contain a
-`<declaringParameterList>`, because that's an error)*.
+A _primary constructor_ declaration consists of a `<primaryConstructor>` in
+the declaration header plus optionally a member declaration in the body
+that starts with a `<primaryConstructorBodySignature>`.
 
-A class or extension type declaration whose class body is `;` is treated as
-a declaration whose body is `{}`.
+A class, mixin class, or extension type declaration whose class body is `;`
+is treated as the corresponding declaration whose body is `{}` and
+otherwise the same.
```

Comment:
(Do we need to say "which is not a `<mixinApplicationClass>` declaration", or say "whose `<classBody>` is `;`", so people don't start thinking `class C = S with M;` is the same as `class C = S with M {}` ... which isn't even valid.)

Reply:
Good catch, done!